rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java

```diff
@@ -0,0 +1,700 @@
+     */
+    private void initializeWebClient() throws RestClientException {
+        if (config.getBaseUrl() != null) {
+            try {
+                new URI(config.getBaseUrl());
+            } catch (URISyntaxException ex) {
+                throw new RestClientException("Invalid parameter baseUrl");
+            }
+        }
+        WebClient.Builder builder = WebClient.builder();
+        HttpClient httpClient = HttpClient.create();
+        SslContext sslContext;
+        try {
+            if (config.isAcceptInvalidSslCertificate()) {
+                sslContext = SslContextBuilder
+                        .forClient()
+                        .trustManager(InsecureTrustManagerFactory.INSTANCE)
+                        .build();
+                httpClient = httpClient.secure(sslContextSpec -> sslContextSpec.sslContext(sslContext));
+            }
+        } catch (SSLException ex) {
+            throw new RestClientException("SSL error occurred: " + ex.getMessage(), ex);
+        }
+        httpClient = httpClient.tcpConfiguration(tcpClient -> {
+                            if (config.getConnectionTimeout() != null) {
+                                tcpClient = tcpClient.option(
+                                        ChannelOption.CONNECT_TIMEOUT_MILLIS,
+                                        config.getConnectionTimeout());
+                            }
+                            if (config.isProxyEnabled()) {
+                                tcpClient = tcpClient.proxy(proxySpec -> {
+                                    ProxyProvider.Builder proxyBuilder = proxySpec
+                                            .type(ProxyProvider.Proxy.HTTP)
+                                            .host(config.getProxyHost())
+                                            .port(config.getProxyPort());
+                                    if (config.getProxyUsername() != null && !config.getProxyUsername().isEmpty()) {
+                                        proxyBuilder.username(config.getProxyUsername());
+                                        proxyBuilder.password(s -> config.getProxyPassword());
+                                    }
+                                    proxyBuilder.build();
+                                });
+                            }
+                            return tcpClient;
+                        }
+                );
+        builder.exchangeStrategies(ExchangeStrategies.builder()
+                .codecs(configurer -> configurer
+                        .defaultCodecs()
+                        .maxInMemorySize(config.getMaxInMemorySize()))
+                .build());
+
+        if (config.getHttpBasicAuthUsername() != null) {
+            builder.filter(ExchangeFilterFunctions
+                    .basicAuthentication(config.getHttpBasicAuthUsername(), config.getHttpBasicAuthPassword()))
+                    .build();
```

Comment:
Do we really want to call `build` here? Note that we are not using the `HttpBasicAuthBuilder` instance here...

Reply:
Not needed, removed.